### PR TITLE
[client-v2] primitive reading into pojo

### PR DESCRIPTION
**Pull request by chernser**
## Summary
Current implementation reads all values into objects even it can be done using primitive types. It was done to simplify code, but it may be not good for memory consumption.
Current PR:
- adds methods to read values into primitives 
- updates POJO deserialization code to use these new methods

## Checklist
Delete items not relevant to your PR:
- [x] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java [hunk 1]
```diff
@@ -91,6 +89,10 @@ public class QueryTests extends BaseIntegrationTest {
 
     private boolean usePreallocatedBuffers = false;
 
+    static {
+//        System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", "DEBUG");
```
Comment (mzitnik) [inline client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java:93]: Lets remove it

**Comment by mzitnik** [comment]
Do we support POJO with types of Long, Integer, Short, and more class

File: client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java (hunk 1 above)
Comment (chernser) [reply client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java:93]: ok. this is only for current development. but will move into env settings on local

**Comment by chernser** [comment]
> Do we support POJO with types of Long, Integer, Short, and more class

We do but with 1-to-1 match. But with primitives it should be easier to handle. 
I will update this part.

**Review by mzitnik — APPROVED** [review]

**Pull request merged**